### ISSUE TITLE
SDL_Init with SDL_INIT_JOYSTICK or SDL_INIT_GAMECONTROLLER stopped working

### ISSUE THREAD

**Issue by zjeffer** [hyprwm/Hyprland#5485]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 3d64b0e9f08151786d354c0ca45c5c6016a16836 dirty (flake.lock: update).
Date: Sat Apr 6 18:09:37 2024
Tag: v0.38.0-54-g3d64b0e9, commits: 4387

flags: (if any)


System Information:
System name: Linux
Node name: zjeffer
Release: 6.8.2-arch2-1
Version: #1 SMP PREEMPT_DYNAMIC Thu, 28 Mar 2024 17:06:35 +0000


GPU information: 
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GA104M [GeForce RTX 3070 Mobile / Max-Q] [10de:24dd] (rev a1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
  split-monitor-workspaces by Duckonaut ver 1.1.0
```

</details>

### Bug or Regression?

Bug

### Description

Since a couple of days, Jellyfin Media Player (a Qt C++ application that uses SDL) has stopped working on Hyprland. Someone else is having the same problem: https://github.com/jellyfin/jellyfin-media-player/issues/589

As I found [here](https://github.com/jellyfin/jellyfin-media-player/issues/589#issuecomment-2041526522), SDL_Init is stuck when calling `SDL_Init(SDL_INIT_JOYSTICK)`. I'm not sure how to debug this further, but I managed to reproduce it with the C++ code below. 

### How to reproduce

```cpp
#include <SDL2/SDL.h>

int main(int argc, char *argv[]) {
  SDL_Log("Initializing SDL");
  if (SDL_Init(SDL_INIT_JOYSTICK) != 0) {
    SDL_Log("Failed to initialize SDL: %s", SDL_GetError());
    return 1;
  }
  SDL_Log("SDL initialized successfully");

  SDL_Quit();
  return 0;
}
```

```cmake
cmake_minimum_required(VERSION 3.12)
project(sdl-test)
set(CMAKE_CXX_STANDARD 23)

find_package(SDL2 REQUIRED)

set(SOURCES src/main.cpp)
add_executable(sdl-test ${SOURCES})
target_link_libraries(sdl-test SDL2)
```

I found that the SDL_Init function freezes the application completely when calling it with SDL_INIT_JOYSTICK or SDL_INIT_GAMECONTROLLER. The other flags work fine.

### Crash reports, logs, images, videos

Backtrace after interrupting the process when it freezes:

```
(gdb) bt
#0  0x00007ffff79196bc in __GI___libc_read (nbytes=19, buf=0x7fffffffcbe0, fd=11) at ../sysdeps/unix/sysv/linux/read.c:26
#1  __GI___libc_read (fd=fd@entry=11, buf=buf@entry=0x7fffffffcbe0, nbytes=nbytes@entry=19) at ../sysdeps/unix/sysv/linux/read.c:24
#2  0x00007ffff742fe32 in read (__nbytes=<optimized out>, __buf=<optimized out>, __fd=<optimized out>, __fd=<optimized out>, 
    __buf=<optimized out>, __nbytes=<optimized out>) at /usr/include/bits/unistd.h:28
#3  read_sysfs_attr (ctx=0x55555558ecc0, sysfs_dir=<optimized out>, attr=attr@entry=0x7ffff743c280 "bConfigurationValue", 
    max_value=max_value@entry=255, value_p=value_p@entry=0x7fffffffcc44) at os/linux_usbfs.c:511
#4  0x00007ffff743008c in sysfs_get_active_config (config=0x7fffffffcc44, dev=0x5555555d2450) at os/linux_usbfs.c:584
#5  op_get_active_config_descriptor (dev=dev@entry=0x5555555d2450, buffer=buffer@entry=0x7fffffffcc9e, len=len@entry=9) at os/linux_usbfs.c:806
#6  0x00007ffff742f14f in get_active_config_descriptor (dev=dev@entry=0x5555555d2450, buffer=buffer@entry=0x7fffffffcc9e "", size=size@entry=9)
    at /usr/src/debug/libusb/libusb-1.0.27/libusb/descriptor.c:483
#7  0x00007ffff7433e33 in libusb_get_active_config_descriptor (dev=0x5555555d2450, config=0x7fffffffcd68)
    at /usr/src/debug/libusb/libusb-1.0.27/libusb/descriptor.c:563
#8  0x00007ffff7de6939 in LIBUSB_hid_enumerate (vendor_id=vendor_id@entry=0, product_id=product_id@entry=0)
    at /usr/src/debug/sdl2/SDL2-2.30.2/src/hidapi/libusb/hid.c:799
#9  0x00007ffff7de6e9b in SDL_hid_enumerate_REAL (vendor_id=0, product_id=0) at /usr/src/debug/sdl2/SDL2-2.30.2/src/hidapi/SDL_hidapi.c:1236
#10 0x00007ffff7ed77a3 in HIDAPI_UpdateDeviceList () at /usr/src/debug/sdl2/SDL2-2.30.2/src/joystick/hidapi/SDL_hidapijoystick.c:1108
#11 0x00007ffff7ed7daf in HIDAPI_JoystickInit () at /usr/src/debug/sdl2/SDL2-2.30.2/src/joystick/hidapi/SDL_hidapijoystick.c:620
#12 0x00007ffff7dbf708 in SDL_JoystickInit () at /usr/src/debug/sdl2/SDL2-2.30.2/src/joystick/SDL_joystick.c:632
#13 SDL_InitSubSystem_REAL (flags=512) at /usr/src/debug/sdl2/SDL2-2.30.2/src/SDL.c:300
#14 0x000055555555519e in main (argc=1, argv=0x7fffffffd148) at /home/zjeffer/git/sdl-test/my-sdl-project/src/main.cpp:5
```

**Comment by vaxerski** [comment]
bisect please

**Comment by zjeffer** [comment]
Can't reproduce after bisecting: the current commit was bad, I rebooted into an older commit that was good, did that a couple of times, then rebooted back into the commit that was supposedly bad and it works fine now.

This is really weird as I had already rebooted once and that didn't fix it. 

I actually had the same exact issue a couple of days ago with Jellyfin Media Player not starting, and it was also resolved after a (single) reboot. Seems to be an intermittent issue? Not sure if it's even Hyprland related. 

I'll reopen once I get the problem again and/or have more info about if it's Hyprland related or not.
